editor/src/scene_viewer/gizmo.rs

```diff
@@ -267,41 +274,53 @@ impl SceneGizmo {
                 }
             }
         }
-
         closest
     }
 
-    pub fn on_mouse_move(&self, pos: Vector2<f32>, engine: &Engine) {
-        let graph = &engine.scenes[self.scene].graph;
-        let closest = self.pick(pos, engine);
-
-        fn set_color(node: Handle<Node>, graph: &Graph, color: Color) {
-            graph[node].as_mesh().surfaces()[0]
-                .material()
-                .data_ref()
-                .set_property(
-                    &ImmutableString::new("diffuseColor"),
-                    PropertyValue::Color(color),
-                )
-                .unwrap();
+    pub fn on_mouse_move(
+        &mut self,
+        pos: Vector2<f32>,
+        engine: &mut Engine,
+        camera_controller: &mut CameraController,
+    ) {
+        if let Some(drag_context) = self.drag_context.as_ref() {
+            let delta = pos - drag_context.initial_click_pos;
+            let sens: f32 = 0.09;
+            camera_controller.yaw = drag_context.initial_rotation.yaw + delta.x * -sens;
+            camera_controller.pitch = drag_context.initial_rotation.pitch + delta.y * sens;
```

Comment:
I completely forgot - here you need to clamp the pitch in `-90.0..90.0` degree range. However, since all angles are in radians, the range will be `-90.0.to_radians()..90.0.to_radians()`. This is why you can rotate the camera upside down using scene gizmo.

Reply:
Great! I can do that. Was also curious about leaving the sens variable there, my intuition was to set it as a const but I haven't noticed many consts in my perusal of the engine. I could also just place the number but I felt that labelling it gave it extra readability. 

 Eventually there could be finer grained controls (allowing the user to turn off dragging, etc etc) of the gizmo so the sensitivity might be good to place into the drag_context as a const.